plugins/coverimage.koplugin/main.lua

```diff
@@ -158,16 +182,413 @@ function CoverImage:onReaderReady(doc_settings)
     self:createCoverImage(doc_settings)
 end
 
+---------------------------
+-- cache handling functions
+---------------------------
+
+function CoverImage:emptyCache()
+    for entry in lfs.dir(self.cover_image_cache_path) do
+        if entry ~= "." and entry ~= ".." then
+            local file = self.cover_image_cache_path .. entry
+            if entry:sub(1,self.cover_image_cache_prefix:len()) == self.cover_image_cache_prefix
+                and lfs.attributes(file, "mode") == "file" then
+                    os.remove(file)
+            end
+        end
+    end
+end
```

Comment:
@zwim 
> Could this be, because I use the koreader/cache folder as default? If yes, which folder can i use safely?

At a quick glance it looks like you're deleting *everything* instead of just your stuff?

Reply:
Maybe I am to "shorteyed":
`entry:sub(1,self.cover_image_cache_prefix:len()) == self.cover_image_cache_prefix` should only select the prefixed files?

Reply:
Provided the prefix is defined properly that'll be fine. But when I read the other message I skimmed quickly and saw this `for entry in lfs.dir(self.cover_image_cache_path) do` … `os.remove`. But I did correctly myself five seconds later. :-P

Reply:
Absolutely no problem. Thanks for your help.

But the underlaying problem consisst: If I use the 'cache' folder I get the error. If i use an other folder, I dont get the error.

Reply:
And the normal cache is working as expected?

Reply:
> And the normal cache is working as expected?
As far as I see yes

I can reproduce the crash on koreader-2021.3: 

1. Copy any file in the `cache` directory.
2. Start koreader, open a book. 
3. Delete jus the file.
4. Open filemanager
5.  crash

Reply:
So, I guess koreader/cache is not just "any" cache, and is expected to be managed only by cache.lua - so we're not free to 't add/remove other stuff.
Although it is fine with subdirectories and the file fontinfo.dat.
So, I guess for your stuff: either in a subdirectory (so, still candidate to the menu item "Clear readers' cache") or elsewhere.

Reply:
I haven't looked deeper in the frontend/cache thing. But I suspect all files in there are checked by the cache mechanism. I one vanishes we get the `nil` exception -> So I will use a subfolder in there as default.

Reply:
It keeps a ... cached (;p) list of entries, so, yeah, it does not expect stuff to mess it.

I guess `fontinfo.dat` kinda works because either it's created early and/or never removed?

Reply:
Ah, a list of *file* entries, actually, which is how CRe gets away with it ;).

I suspect that if you tried to delete `fontinfo.dat`, it would *also* be upset.

Reply:
Yes, yes, it does. I'll fix fontlist not to do that.

Reply:
Or maybe move fontlist.dat into koreader/settings/ ? It's some KOReader UI font cache, it does not really belong among "reader's caches" :)

Reply:
Well, it *is*, semantically, a cache, though...